### ISSUE TITLE
[Gateway] Splunk-otel-collector overwrides host dimension

### ISSUE THREAD

**Issue by AchimGrolimund** [signalfx/splunk-otel-collector#689]
Hello all

I have the situation that i have an SignalFx-agent installed on an onPremise Host. they send datas to the Gateway (Splunk-otel-collector --mode gateway).

in signalfx i recive all datas, but I see that the gateway overwrites many dimensions and or no longer shows in SignalFx.

Dataflow:
onPremise Signalfx-Agent --> Gateway on AWS (Splunk Otel-colector) --> SignalFX

Hier the Configs:

**onpremise SignalFX-Agent:**
```yaml
---
# *Required* The access token for the org that you wish to send metrics to.
signalFxAccessToken: {"#from": 'c:\ProgramData\SignalFxAgent\token'}
ingestUrl: {"#from": 'c:\ProgramData\SignalFxAgent\ingest_url', default: "https://xxxxxxx/ingest/"}
apiUrl: {"#from": 'c:\ProgramData\SignalFxAgent\api_url', default: "https://xxxxxxxx/api/"}
eventEndpointUrl:  {"#from": 'c:\ProgramData\SignalFxAgent\event_url', default: "https://xxxxxxxxx/ingest/v2/event"}

intervalSeconds: 10

logging:
  # Valid values are 'debug', 'info', 'warning', and 'error'
  level: info

# observers are what discover running services in the environment
observers:
  - type: host

globalDimensions: 
  {"#from": 'C:\ProgramData\SignalFxAgent\globalDimensions\*.yaml', optional: true}

monitors:
  - {"#from": 'c:\ProgramData\SignalFxAgent\monitors\*.yaml', flatten: true, optional: true}
```

**GatewayConf:**
```yaml
# Configuration file that uses the Splunk exporters (SAPM, SignalFx) to push
# data to Splunk products.

extensions:
  health_check:
    endpoint: 0.0.0.0:13133
  http_forwarder:
    ingress:
      endpoint: 0.0.0.0:6060
    egress:
      endpoint: "https://api.${SPLUNK_REALM}.signalfx.com"
  zpages:
    endpoint: 0.0.0.0:55679
  memory_ballast:
    # In general, the ballast should be set to 1/3 of the collector's memory, the limit
    # should be 90% of the collector's memory.
    # The simplest way to specify the ballast size is set the value of SPLUNK_BALLAST_SIZE_MIB env variable.
    size_mib: ${SPLUNK_BALLAST_SIZE_MIB}

receivers:
  hostmetrics:
    collection_interval: 10s
    scrapers:
      cpu:
      disk:
      filesystem:
      memory:
      network:
  jaeger:
    protocols:
      grpc:
        endpoint: 0.0.0.0:14250
      thrift_binary:
        endpoint: 0.0.0.0:6832
      thrift_compact:
        endpoint: 0.0.0.0:6831
      thrift_http:
        endpoint: 0.0.0.0:14268
  otlp:
    protocols:
      grpc:
        endpoint: 0.0.0.0:4317
      http:
        endpoint: 0.0.0.0:55681
  # This section is used to collect the OpenTelemetry Collector metrics
  # Even if just a Splunk APM customer, these metrics are included
  prometheus/internal:
    config:
      scrape_configs:
        - job_name: 'otel-collector'
          scrape_interval: 10s
          static_configs:
            - targets: ['0.0.0.0:8888']
          metric_relabel_configs:
            - source_labels: [ __name__ ]
              regex: '.*grpc_io.*'
              action: drop
  sapm:
    endpoint: 0.0.0.0:7276
  signalfx:
    endpoint: 0.0.0.0:9943
  zipkin:
    endpoint: 0.0.0.0:9411

processors:
  batch:
  # Enabling the memory_limiter is strongly recommended for every pipeline.
  # Configuration is based on the amount of memory allocated to the collector.
  # For more information about memory limiter, see
  # https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/memorylimiter/README.md
  memory_limiter:
    check_interval: 2s
    limit_mib: ${SPLUNK_MEMORY_LIMIT_MIB}

      # Optional: The following processor can be used to add a default "deployment.environment" attribute to the traces
      # when it's not populated by instrumentation libraries.
      # If enabled, make sure to enable this processor in the pipeline below.
      #resource/add_environment:
      #attributes:
    #- action: insert
    #value: staging/production/...
    #key: deployment.environment

  # detect if the collector is running on a cloud system.
  # overrides resource attributes set by receivers
  resourcedetection/internal:
    detectors: [system, gce, ecs, ec2, azure]
    override: true

  # detect if the collector is running on a cloud system
  # important for creating unique cloud provider dimensions
  resourcedetection:
    detectors: [system, gce, ecs, ec2, azure]
    override: false

exporters:
  # Traces
  sapm:
    access_token: "${SPLUNK_ACCESS_TOKEN}"
    endpoint: "https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace"
  # Metrics + Events
  signalfx:
    access_token: "${SPLUNK_ACCESS_TOKEN}"
    realm: "${SPLUNK_REALM}"
    ## Uncomment below if your agents are sending via signalfx exporter
    ## to avoid double translations and exclusions.
    translation_rules: []
    exclude_metrics: []
    # Debug
    #logging:
    #loglevel: debug

service:
  extensions: [health_check, http_forwarder, zpages, memory_ballast]
  pipelines:
    traces:
      receivers: [jaeger, otlp, sapm, zipkin]
      processors:
        - memory_limiter
        - batch
      #- resource/add_environment
      exporters: [sapm, signalfx]
    metrics:
      receivers: [hostmetrics, otlp, signalfx]
      processors: [memory_limiter, batch, resourcedetection]
      exporters: [signalfx]
    metrics/internal:
      receivers: [prometheus/internal]
      processors: [memory_limiter, batch, resourcedetection/internal]
      exporters: [signalfx]
    logs:
      receivers: [signalfx]
      processors: [memory_limiter, batch]
      exporters: [signalfx]
```

![image](https://user-images.githubusercontent.com/22026079/131501078-f5376d6e-abd5-447a-86df-81f0772935df.png)

**In short, the hostname should be that of the on-premise host where the agent is running and not the hostname of the gateway**

i try also to switch the `override: ` from the resurcedetection from `true` to `false` and vice versa

Best Regards

**Comment by mstumpfx** [comment]
The `resourcedetection` processor is adding the this dimension to your datapoints via the `env` detection. 
If you are trying to monitor the gateway itself, I would move the `hostmetrics` to the `metrics/internal` service pipeline instead of the `metrics` pipeline, and remove the `resourcedection` from the metrics pipeline.
This way you will have your metrics being forwarded pass through as-is, and the internal metrics reported about the gateway itself will still use the `resourcedetection` processor to add the host dimension.
e.g.

```
service:
  ...
  pipelines:
    ...
    metrics:
      receivers: [otlp, signalfx]
      processors: [memory_limiter, batch]
      exporters: [signalfx]
    metrics/internal:
      receivers: [prometheus/internal, hostmetrics]
      processors: [memory_limiter, batch, resourcedetection/internal]
      exporters: [signalfx]
```

**Comment by AchimGrolimund** [comment]
hey @mstumpfx 

Thank you for your answer!
I have just adjusted the config and am testing it out.

yes, you recognized it correctly, I would also like to monitor the gateway myself. and it doesn't make sense to run a connector as a gateway and as an agent on a host.

**Comment by AchimGrolimund** [comment]
@mstumpfx 

but on your example i dont see anymore the metrics from the Gateway host its self
![image](https://user-images.githubusercontent.com/22026079/131509449-2efd7bf0-c0df-4bb3-bce4-4c401be58044.png)

**Comment by mstumpfx** [comment]
Can you try changing your filter on the chart to use `host.name` instead of `host`? 

We have something running in our backend what we call "mapping service". This service maps metrics and dimensions to other metric names and dimensions. For example, we map `host` : `host.name`. We do this because the original signalfx-agent sent the `host` dimension, but the splunk-otel-collector sends the `host.name` dimension. This service is meant to help users migrate agents, but we sometimes run into small issues like this along the way. Some related reading: https://docs.signalfx.com/en/latest/getting-started/concepts/mapping-transition-rept.html 
List of metrics / dimensions we map: https://github.com/signalfx/integrations/blob/mappings/mappings/mappings.md

**Comment by AchimGrolimund** [comment]
Hey @mstumpfx 
Thanks again

same result and the same count of TS 
![image](https://user-images.githubusercontent.com/22026079/131511477-cc4e9d5c-934b-40bb-8c4f-15776c32c853.png)

**Comment by AchimGrolimund** [comment]
i see there is a mapping also from my Metric what i search.

when i search system.filesystem.usage then i can see som informations.

but this metric is not listet on the table on https://github.com/signalfx/integrations/blob/mappings/mappings/mappings.md

**Comment by AchimGrolimund** [comment]
Does this mean that as soon as I replace the SignalFx agents with the new Splunk-Otel-Collector, do I have to readjust all the charts because of the renaming of the metrics? Or I do these calculations, but have custom metrics for them

**Comment by mstumpfx** [comment]
Sorry for the delayed response... 
Mapping service should cover you for the metric renames, or the the default translation rules I linked above will export the metrics. There should not be any renaming of charts needed.
